chat-commands.js

```diff
@@ -2503,47 +2511,105 @@ exports.commands = {
 		"/hotpatch dnsbl - reloads Dnsbl datacenters",
-		"/hotpatch disable, [reason] - disables the use of hotpatch until the next server restart"],
+		"/hotpatch punishments - reloads new punishments code",
+		"/hotpatch tournaments - reloads new tournaments code",
+	],
+
+	hotpatchlock: 'disablehotpatch',
+	nohotpatch: 'disablehotpatch',
+	hotpatchdisable: 'disablehotpatch',
+	disablehotpatch: function (target, room, user) {
+		if (!this.can('hotpatch')) return;
+		if (!target) return this.parse('/help disablehotpatch');
+
+		const separator = ' ';
+
+		const hotpatch = target.substr(0, target.indexOf(separator)).trim();
+		const reason = target.substr(target.indexOf(separator), target.length).trim();
+		if (!reason || !target.includes(separator)) return this.parse('/help disablehotpatch');
+
+		let lock = Monitor.hotpatchLock;
+
+		if (hotpatch === 'chat') {
+			if (lock['chat']) return this.errorReply(`Hot-patching chat has already been disabled by ${lock['chat'].by} (${lock['chat'].reason})`);
+			lock['chat'] = {
+				by: user.name,
+				reason: reason,
+			};
+			this.sendReply("You have disabled hot-patching chat.");
+		} else if (hotpatch === 'battles') {
+			if (lock['battles']) return this.errorReply(`Hot-patching battles has already been disabled by ${lock['battles'].by} (${lock['battles'].reason})`);
+			lock['battles'] = {
+				by: user.name,
+				reason: reason,
+			};
+			this.sendReply("You have disabled hot-patching battles.");
+		} else if (hotpatch === 'formats') {
+			if (lock['formats']) return this.errorReply(`Hot-patching formats has already been disabled by ${lock['formats'].by} (${lock['formats'].reason})`);
+			lock['formats'] = {
+				by: user.name,
+				reason: reason,
+			};
+			this.sendReply("You have disabled hot-patching formats.");
+		} else if (hotpatch === 'validator' || hotpatch === 'learnsets') {
+			if (lock['validator']) return this.errorReply(`Hot-patching the validator has already been disabled by ${lock['validator'].by} (${lock['validator'].reason})`);
+			lock['validator'] = {
+				by: user.name,
+				reason: reason,
+			};
+			this.sendReply("You have disabled hot-patching the validator.");
+		} else if (hotpatch === 'tournaments' || hotpatch === 'tours') {
+			if (lock['tournaments']) return this.errorReply(`Hot-patching tournaments has already been disabled by ${lock['tournaments'].by} (${lock['tournaments'].reason})`);
+			lock['tournaments'] = {
+				by: user.name,
+				reason: reason,
+			};
+			this.sendReply("You have disabled hot-patching tournaments.");
+		} else {
+			return this.errorReply("This hotpatch is not an option to disable.");
+		}
+		Rooms.global.notifyRooms(['development', 'staff', 'upperstaff'], `|c|${user.getIdentity()}|/log ${user.name} has disabled hot-patching ${hotpatch}.`);
+	},
+	disablehotpatchhelp: ["/disablehotpatch [chat|formats|battles|validator|tournaments] [reason] - Disables hotpatching the specified part of the simulator. Requires: ~"],
```

Comment:
By which I mean, support both `/hotpatch all` and `/nohotpatch all`; we used to have `/hotpatch all` but removed it because it did too much damage to Main.